protobufs/livekit_models.proto

```diff
@@ -601,3 +601,60 @@ enum AudioTrackFeature {
   TF_ENHANCED_NOISE_CANCELLATION = 5;
 }
 
+message DataStream {
+
+  // enum for operation types (specific to TextStreamHeader)
+  enum OperationType {
+    CREATE = 0;
+    UPDATE = 1;
+    DELETE = 2;
+    REACTION = 3;
+  }
+  
+  // header properties specific to text streams
+  message TextHeader {
+    OperationType operationType = 1;
+    int32 version = 2;  // Optional: Version for updates/edits
+    string replyToStreamId = 3;  // Optional: Reply to specific message
+    repeated string attachedStreamIds = 4;
+    bool generated = 5; // true if the text has been generated by an agent from a participant's audio transcription
+
+  }
+  
+  // header properties specific to file or image streams
+  message FileHeader {
+    string fileName = 1;  // optional: name of the file
+  }
+
+  // main DataStreamHeader that contains a oneof for specific headers
```

Comment:
:nit: maybe change comment to say `DataStream.Header` to signify that it is a not a type of its own.